File: apps/merchant-api/InventoryService/src/InventoryService.Domain/Concretes/Variant.cs
using InventoryService.Domain.Bases;

namespace InventoryService.Domain.Concretes;

public class Variant : BaseEntity
{ 
    public string Name { get; set; } = string.Empty;
    public ICollection<ProductAttribute> ProductAttributes { get; set; } = new List<ProductAttribute>();    
}

File: apps/merchant-web/src/commons/entities/concretes/Variant.ts
import { UUID } from "crypto";
import EntityBase from "../EntityBase";
import ProductAttribute from "./ProductAttribute";

export default class Variant extends EntityBase {
  name: string;
  productAttributes: Array<ProductAttribute>;

  constructor(id: UUID, name: string, productAttributes: Array<ProductAttribute>) {
    super(id);
    this.name = name;
    this.productAttributes = productAttributes;
  }
}